Extensions/BitmapText/bitmaptextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,209 @@
+  runtimeObject,
+  runtimeScene
+) {
+  this._object = runtimeObject;
+
+  // Set up the object to track the font we're using.
+  this._bitmapFontStyle = new PIXI.TextStyle();
+  this._bitmapFontStyle.fontFamily = 'Arial';
+  this._bitmapFontStyle.fontSize = runtimeObject._fontSize;
+  this._bitmapFontStyle.align = runtimeObject._align;
+  this._bitmapFontStyle.wordWrap = runtimeObject._wordWrap;
+  this._bitmapFontStyle.fill = gdjs.rgbToHexNumber(
+    runtimeObject._fontColor[0],
+    runtimeObject._fontColor[1],
+    runtimeObject._fontColor[2]
+  );
+
+  const defaultSlugFontName =
+    this._bitmapFontStyle.fontFamily +
+    '-' +
+    this._bitmapFontStyle.fontSize +
+    '-' +
+    this._bitmapFontStyle.fill +
+    '-bitmapFont';
+  this._bitmapFontStyle.fontName = defaultSlugFontName;
+
+  // Load (or reset) the text
+  if (this._pixiObject === undefined) {
+    //Generate default bitmap font
+    PIXI.BitmapFont.from(defaultSlugFontName, this._bitmapFontStyle, {
+      chars: [
+        [' ', '~'], // All the printable ASCII characters
+      ],
+    });
+
+    this._pixiObject = new PIXI.BitmapText(runtimeObject._text, {
+      fontName: defaultSlugFontName,
+    });
+
+    let bitmapFontResourceName = runtimeObject._bitmapFontFile;
+    let bitmapTextureResourceName = runtimeObject._bitmapTextureFile;
+
+    let texture = runtimeScene
+      .getGame()
+      .getImageManager()
+      .getPIXITexture(bitmapTextureResourceName);
+
+    // Get the bitmap font file and use the texture for generate the bitmapFont (PIXI.BitmapFont) and return the fontName ready to use.
+    runtimeScene
+      .getGame()
+      .getBitmapFontManager()
+      .loadBitmapFont(bitmapFontResourceName, texture)
+      .then((fontName) => {
+        this._pixiObject.fontName = fontName;
+        this.updatePosition();
+      })
+      .catch((error) => {
+        console.error('Error while loading a bitmapFont resource:', error);
+      });
+  } else {
+    this.updateFont();
+    this.updateFontSize();
+  }
+
+  runtimeScene
+    .getLayer('')
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  // Set the anchor in the center, so that the object rotates around
+  // its center.
+  this._pixiObject.anchor.x = 0.5;
+  this._pixiObject.anchor.y = 0.5;
+
+  this.updateAlignment();
+  this.updateTextContent();
+  this.updatePosition();
+  this.updateAngle();
+  this.updateOpacity();
+  this.updateWrappingWidth();
+};
+
+gdjs.BitmapTextRuntimeObjectRenderer = gdjs.BitmapTextRuntimeObjectPixiRenderer;
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  return this._pixiObject;
+};
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype.onDestroy = function () {
+  // Mark the font from the object not used anymore.
+  gdjs.BitmapFontManager.removeFontUsed(this._pixiObject._fontName);
+};
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype._ensureFontAvailableAndGetFontName = function (
```

Comment:
Is this font handling still necessary with the new fnt/xml bitmap font resources?
Or is it a fallback in case you don't specify a fnt/xml file?